### PR TITLE
Add Pretrained TTS test script; Extend from_pretrained() Functionality

### PR DESCRIPTION
**Pull request by blisc**
This PR does the following

- Add a new TTS -> ASR using pretrained Tacotron2, Waveglow, and QuartzNet models as a regression CI test
- Updates tacotron such that it masking with silence past predicted lengths
- Changes PretrainedModelInfo to add a new optional field class_
- If class_ is defined inside PretrainedModelInfo, from_pretrained() now instantiates that class_ instead of cls. IE, so long as Model has PretrainedModelInfo with class_, one can now do Model.from_pretrained("my_model") instead of my_model.from_pretrained("my_model")

### CODE REVIEW

File: examples/asr/speech_to_text_infer.py [hunk 1]
```diff
@@ -46,7 +46,6 @@ def main():
         "--asr_model", type=str, default="QuartzNet15x5Base-En", required=True, help="Pass: 'QuartzNet15x5Base-En'",
     )
     parser.add_argument("--dataset", type=str, required=True, help="path to evaluation data")
-    parser.add_argument("--wer_target", type=float, default=None, help="used by test")
```
Comment (titu1994) [inline examples/asr/speech_to_text_infer.py:49]: Why is this being removed?

File: nemo/collections/tts/models/base.py [hunk 2]
```diff
@@ -39,6 +41,18 @@ def generate_spectrogram(self, tokens: 'torch.tensor', **kwargs) -> 'torch.tenso
...
+    @classmethod
+    def list_available_models(cls) -> 'List[PretrainedModelInfo]':
+        """
+        This method returns a list of pre-trained model which can be instantiated directly from NVIDIA's NGC cloud.
+        Returns:
+            List of available pre-trained models.
+        """
+        list_of_models = []
+        for subclass in cls.__subclasses__():
+            list_of_models.extend(subclass.list_available_models())
```
Comment (titu1994) [inline nemo/collections/tts/models/base.py:53]: This is fine here, but subclasses can potentially return None (that's the default), so it should be checked here before extend.

File: nemo/collections/tts/models/base.py [hunk 3]
```diff
@@ -54,3 +68,15 @@ def convert_spectrogram_to_audio(self, spec: 'torch.tensor', **kwargs) -> 'torch
...
```
Comment (titu1994) [inline nemo/collections/tts/models/base.py:81]: Same as above

**Review by titu1994 — COMMENTED** [review]
Looks good overall, but subclass tracing is very hackish, and depends on the assumption that you have flat subclass hierarchy. It's a reasonable solution for the moment however.

File: examples/asr/speech_to_text_infer.py (hunk 1 above)
Comment (blisc) [reply examples/asr/speech_to_text_infer.py:49]: It's not being used. `wer_tolerance` is the correct argument

File: nemo/collections/tts/models/base.py (hunk 2 above)
Comment (blisc) [reply nemo/collections/tts/models/base.py:53]: That is a fair point, will update.

**Review by okuchaiev — APPROVED** [review]
Thanks!
